tinygrad/realize.py

```diff
@@ -12,7 +12,8 @@ def run_schedule(schedule:List[ScheduleItem], disable_logging=False):
     si = schedule.pop(0)
     if not disable_logging: log_schedule_item(si)
     assert all(x.realized for x in si.inputs), "can't run schedule, some inputs aren't realized"
-    assert all(si.out.device == x.device for x in si.inputs) or si.ast.op is LoadOps.FROM, f"all devices must be the same, {si.out.device} != {[x.device for x in si.inputs]} {print_tree(si.ast) or ''}"
+    metal_disk_support = Device.DEFAULT == "METAL" and si.out.device.startswith("DISK:")
```

Comment:
you can combine DISK tensors with METAL tensors and the code will know what to do. not sure if i even like this